server/clients/jsonApiClient.js

```diff
@@ -0,0 +1,35 @@
+const axios = require('axios');
+const httpAdapter = require('axios/lib/adapters/http');
+const { logger } = require('../utils/logger');
+
+class JsonApiClient {
+  constructor(baseURL) {
+    this.client = axios.create({
+      baseURL,
+      timeout: 30000,
+      adapter: httpAdapter,
```

Comment:
did this to make life a bit easier - in tests, axios detects js-dom and uses a fake browser based xmlhttprequest object which doesn't play nice with jest and is not representative as to how the client would work in production.

explicitly specifying the adapter in the prod code stops that from happening